mlir/lib/Dialect/SPIRV/IR/SPIRVOps.cpp

```diff
@@ -1021,6 +1021,25 @@ LogicalResult spirv::FuncOp::verifyType() {
 
 LogicalResult spirv::FuncOp::verifyBody() {
   FunctionType fnType = getFunctionType();
+  if (!isExternal()) {
+    Block &entryBlock = front();
+
+    unsigned numArguments = this->getNumArguments();
+    if (entryBlock.getNumArguments() != numArguments)
+      return emitOpError("entry block must have ")
+             << numArguments << " arguments to match function signature";
+
+    ArrayRef<Type> fnInputTypes = getArgumentTypes();
+    for (unsigned i = 0, e = numArguments; i != e; ++i) {
+      Type argType = entryBlock.getArgument(i).getType();
```

Comment:
I think you should be able to use `for (auto [index, fnArgType, blockArgType] : llvm::enumerate(getArgumentType(), entryBlock.getArgumentTypes()))`